include/sys/arch_interface.h

```diff
@@ -373,7 +374,7 @@ void arch_irq_offload(irq_offload_routine_t routine, const void *parameter);
  */
 #ifdef CONFIG_SMP
 /** Return the CPU struct for the currently executing CPU */
-static inline struct _cpu *arch_curr_cpu(void);
+static ALWAYS_INLINE __attribute__((pure, returns_nonnull)) _cpu_t *arch_curr_cpu(void);
```

Comment:
I'm curious about why do we add these attributes to the `arch_curr_cpu` declaration only and don't touch `arch_curr_cpu` definitions?